<launch>
    <env name="ROS_IP" value="192.168.43.76"/>

    <!-- Launch TagSLAM with respawn -->
    <include file="$(find tagslam)/launch/tagslam_respawn.launch">
    <arg name="run_online" value="true"/> 
    <arg name="tagslam_config_file" value="$(find behaviortree_ros)/config/mission.yaml"/>
    <arg name="output_type" value="log"/>
    <arg name="respawn" value="true"/>
    <arg name="respawn_delay" value="0.1"/>
    </include>
    <!-- Launch apriltag_detector_node.launch -->
    <include file="$(find tagslam)/launch/apriltag_detector_node.launch"/>


    <!-- Launch Behavior Tree waypoint node with waypoints from YAML -->
    <node name="my_waypoint_bt_node" 
    pkg="behaviortree_ros" 
    type="waypoint_bt_node" 
    output="screen">
        <rosparam file="$(find behaviortree_ros)/config/waypoints.yaml" command="load"/>
    </node>


    <!-- Launch trajGen node -->
    <!-- <param name="trajGen/velo" value="0.1"/> -->
    <!-- <node pkg="behaviortree_ros" type="trajgen_supervisor" name="trajgen_supervisor" output="screen">
    </node> -->

    <!-- <node pkg="trajgen" type="trajGen" name="trajGen" output="screen">
    <param name="/velo" value="0.1"/>
    </node> -->


    <param name="/trajGen/velo" value="0.1"/>
    <node pkg="trajgen_with_yaw" type="trajGen_with_yaw" name="trajGen_with_yaw" output="screen">
    </node>
    

    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch"/>
    <node pkg="web_video_server" type="web_video_server" name="web_video_server" output="screen">
        <param name="port" value="9092"/>
    </node>
    <!-- Launch RViz -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find tagslam)/example/tagslam_example.rviz" /> -->
</launch>
